### PR TITLE
Fixes #16: Setup CI for linux and osx

### LINKED ISSUE #16: Continuous Integration
Seems like more and more tutorials are coming soon, including some functionalities from `contrib` module. For example, some people are testing `TF.Learn` out by following its [tutorials](https://www.tensorflow.org/versions/r0.11/tutorials/tflearn/index.html). See https://github.com/rstudio/tensorflow/issues/14

It would be good to setup continuous integration for the repo so some tutorials will be tested against master branch of TF since `contrib` interface will be changing quite often until it's merged into core.

### PR DESCRIPTION
**Pull request by terrytangyuan**
Just had a chance to quickly spin this up. I know `r-travis` is deprecated but I don't have too much time switching to use native R support in Travis at this moment.

### CODE REVIEW

File: run_tests.sh [hunk 1]
```diff
@@ -1,3 +1,3 @@
 #!/bin/bash
 
-Rscript -e "library(tensorflow)"
\ No newline at end of file
+Rscript -e 'Sys.setenv(TENSORFLOW_PYTHON=system("which python", intern = T));library(tensorflow)'
```
Comment (jjallaire) [inline run_tests.sh]: Setting this variable only has an impact prior to configuration/installation not before loading of the library (i.e. I wouldn't expect it to fix anything). I'll remove this after merging and see what happens.

**Comment by jjallaire** [comment]
@eddelbuettel I'm attempting to do a git clone of the repo so I can run the tests but I get this:

```
Cloning into 'tensorflow'...
The authenticity of host 'github.com (192.30.253.112)' can't be established.
RSA key fingerprint is 16:27:ac:a5:76:28:2d:36:63:1b:56:4d:eb:df:a6:48.
Are you sure you want to continue connecting (yes/no)? 
No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
```

Have you seen this before? If so is there a straightforward workaround?

**Comment by terrytangyuan** [comment]
Try using https to clone. I haven't used Travis for a while, but it might have also cloned it to workspace already.

**Comment by eddelbuettel** [comment]
Interesting.  

I forked the r-travis repo by Craig (to which I was an early contributor) as I like the freedom it gives to use other repos (ie PPAs with other prebuild .deb packages), and I am using that over all my repos (incl our Rcpp) via a still fairly close frontend script `run.sh` (but which does not whine _deprecated...).  

As such I basically _never_ directly clone directly from git (but rather access packages from a PPA).   But as @terrytangyuan just said, this looks like the standard `ssh` message you get when not using https and switching to `https` should fix that,

**Comment by jjallaire** [comment]
I sorted it out. There was as you said no need to clone (the script runs in
the context of the repo's directory).

On Sun, Oct 2, 2016 at 10:49 AM, Dirk Eddelbuettel <notifications@github.com

**Comment by jjallaire** [comment]
There still appears to be problems with the tensorflow package loading (it complains about an inadequate version of libc). R CMD check also yields a ton of errors and warnings regarding the binaries which end up getting installed in testenv (I took care of that by adding testenv to .Rbuildignore)

https://travis-ci.org/rstudio/tensorflow/jobs/164440045

I don't have the expertise to debug why tensorflow isn't loading so I'm going to disable travis for now. Welcome any help in troubleshooting you can provide.

**Comment by eddelbuettel** [comment]
Sorry, preoccupied with a cleanup in another package which CRAN requested....

From glancing at [the .travis.yml](https://github.com/rstudio/tensorflow/blob/master/.travis.yml) you _may_ just be missing _one_ magic line:  

```
dist: trusty
```

Travis is a little, err, "funny" in insisting on Ubuntu precise aka 12.04 as the default.  The [joke about not being to send a fax](https://twitter.com/mc_hankins/status/538811462475853824) applies.  I set everything to Ubuntu trusty aka 14.04 which is _also_ outdated but at least not quite _that_ badly.

**Comment by jjallaire** [comment]
Okay, just added that and will try again now.

On Sun, Oct 2, 2016 at 11:39 AM, Dirk Eddelbuettel <notifications@github.com

**Comment by jjallaire** [comment]
Now on Linux I am failing with this:

```
* checking whether package ‘tensorflow’ can be installed ... ERROR
Installation failed.
See ‘/tmp/RtmpA824et/tensorflow.Rcheck/00install.out’ for details.
* DONE
Status: 1 ERROR
See
  ‘/tmp/RtmpA824et/tensorflow.Rcheck/00check.log’
for details.
```

@eddelbuettel or @terrytangyuan  Do you know how I can discover what's in 00check.log to troubleshoot further?

**Comment by eddelbuettel** [comment]
I usually end the `.travis.yml` on 

```
after_failure:
  - ./travis-tool.sh dump_logs

notifications:
  email:
    on_success: change
    on_failure: change
```

and you may just need the `after_failure` condition.  The log is then accessible in the web front-end (and be copied out from there).

**Pull request merged**